### ISSUE TITLE
FR: WebIDL support

### ISSUE THREAD

**Issue by rngadam** [swig/swig#1253]
Emscripten uses WebIDL (deprecating Embind):

* https://heycam.github.io/webidl/
* https://kripken.github.io/emscripten-site/docs/porting/connecting_cpp_and_javascript/WebIDL-Binder.html

```
The WebIDL Binder provides a simple and lightweight approach to binding C++, so that compiled code can be called from JavaScript as if it were a normal JavaScript library.
```

...requires to manually create interface files. I kind of though this would be something SWIG supports.

**Comment by wsfulton** [comment]
SWIG supports the language modules that have been contributed by the users requiring support in a language for wrapping C/C++. SWIG already has 3 Javascript modules. If none of these are suitable, then an explanation is needed why another is required. A volunteer also needs to also implement it!

**Comment by ojwb** [comment]
> If none of these are suitable, then an explanation is needed why another is required. A volunteer also needs to also implement it!

Approaching 4 years on, nobody's explained why another is needed, and nobody's volunteered to implement it so I'm going to close this issue.

If someone explained why WebIDL support was useful and submitted a good quality patch for it which fulfilled the criteria listed in the manual it would still be considered of course.

**Comment by devshgraphicsprogramming** [comment]
WebIDL is not usable at all because it can only express what WebIDL can, no shared pointers, etc.

**Comment by devshgraphicsprogramming** [comment]
also I havent heard anything about deprecating embind.

**Comment by mmomtchev** [comment]
I have published `node-magickwand@1.0.0-beta.0`, you can install it with

```
npm install node-magickwand@next
```

It includes a WASM version built from the same source files as the Node.js native version.

You can read more about SWIG Node-API WASM on the project's homepage.
